File: library/src/main/java/cn/maitian/app/library/LibraryApplication.java
package cn.maitian.app.library;

import android.app.Application;

import com.facebook.stetho.Stetho;

/**
 * Created by IntelliJ IDEA in MaitianCommonLibrary
 * cn.maitian.app.library
 *
 * @Author: Chris Chen
 * @Time: 2019/4/13 13:50
 * @Description:
 */
public class LibraryApplication extends Application {

    private static LibraryApplication _instance;

    private void initStetho () {
        Stetho.initialize(
                Stetho.newInitializerBuilder(this)
                        .enableDumpapp(Stetho.defaultDumperPluginsProvider(this))
                        .enableWebKitInspector(Stetho.defaultInspectorModulesProvider(this))
                        .build());
    }

    public static LibraryApplication getInstance () {
        return _instance;
    }

    @Override
    public void onCreate () {
        super.onCreate();
        _instance = this;

        initStetho();
    }
}

File: library/src/main/java/ie/chrischen/library/LibraryApplication.kt
package ie.chrischen.library

import android.app.Application
import android.content.Context
import android.content.ContextWrapper
import com.facebook.stetho.Stetho

class LibraryApplication : Application() {

    override fun onCreate() {
        super.onCreate()
        Stetho.initialize(
            Stetho.newInitializerBuilder(this)
            .enableDumpapp(Stetho.defaultDumperPluginsProvider(this))
            .enableWebKitInspector(Stetho.defaultInspectorModulesProvider(this))
            .build())
    }

    companion object {
        @JvmStatic
        lateinit var INSTANCE: LibraryApplication
    }
}

object AppContext : ContextWrapper(LibraryApplication.INSTANCE)